src/main/java/com/github/terma/jenkins/githubprcoveragestatus/CompareCoverageAction.java

```diff
@@ -74,10 +87,29 @@ public void perform(
             return;
         }
 
+
         buildLog.println(BUILD_LOG_PREFIX + "start");
 
-        final String gitUrl = Utils.getGitUrl(build, listener);
-        final int prId = Utils.gitPrId(build, listener);
+        String gitUrl;
+        int prId;
+        try {
```

Comment:
Please replace that ```try catch``` logic on simple ```if``` for example:
```
if (Utils.getPrId(build, listener) != null) // logic
else if (scmVars != null && scmVars.contains(...)) // logic
else throw new IAE("Please provide PR_ID or scmVars");
``` 
That's simplify logic and you can avoid additional settings ```isPrDiscoveryForBranches```